### PR TITLE
DMP-2884 ARM Synchronous Deletion - IU file deletion

### PR DESCRIPTION
**Pull request by karen-hedges**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/DMP-2884

### Change description ###

Added functionality to delete the batch IU file

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ x] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/darts/arm/service/impl/ArmBatchProcessResponseFilesImpl.java [hunk 1]
```diff
@@ -166,7 +203,7 @@ private void resetArmStatusForUnprocessedEods(String manifestName) {
         }
     }
 
-    private void processResponseFileByHashcode(BatchInputUploadFileFilenameProcessor batchUploadFileFilenameProcessor) {
+    private void processResponseFileByHashcode(BatchInputUploadFileFilenameProcessor batchUploadFileFilenameProcessor, String manifestName) {
```
Comment (johnrace0) [inline src/main/java/uk/gov/hmcts/darts/arm/service/impl/ArmBatchProcessResponseFilesImpl.java:206]: is this new param needed?

**Review by chrisbellingham-hmcts — COMMENTED** [review]
Are there any tests (unit/int) that can be written to verify the new functionality?

File: src/main/java/uk/gov/hmcts/darts/arm/service/impl/ArmBatchProcessResponseFilesImpl.java (hunk 1 above)
Comment (karen-hedges) [reply src/main/java/uk/gov/hmcts/darts/arm/service/impl/ArmBatchProcessResponseFilesImpl.java:206]: Its used for logging. There is an issue with tracing log messages via dynatrace as it uses different things in the logs.

**Comment by karen-hedges** [comment]
> Are there any tests (unit/int) that can be written to verify the new functionality?

I have updated the tests to cover this

File: src/main/java/uk/gov/hmcts/darts/arm/service/impl/ArmBatchProcessResponseFilesImpl.java (hunk 1 above)
Comment (chrisbellingham-hmcts) [reply src/main/java/uk/gov/hmcts/darts/arm/service/impl/ArmBatchProcessResponseFilesImpl.java:206]: The `manifestName` parameter isn't used anywhere in the `processResponseFileByHashcode()` method though?

Comment (karen-hedges) [reply src/main/java/uk/gov/hmcts/darts/arm/service/impl/ArmBatchProcessResponseFilesImpl.java:206]: It is on line 221 in the log message

**Pull request merged**